src/net/atlas/db.rs

```diff
@@ -231,30 +257,76 @@ impl AtlasDB {
         Ok(res)
     }
 
-    pub fn insert_new_attachment(&mut self, attachment: &Attachment) -> Result<(), db_error> {
+    pub fn insert_uninstanciated_attachment(
+        &mut self,
+        attachment: &Attachment,
+    ) -> Result<(), db_error> {
+        // Insert the new attachment
+        let uninstantiated_attachments = self.count_uninstantiated_attachments()?;
+        if uninstantiated_attachments >= self.atlas_config.max_uninstanciated_attachments {
+            let to_delete =
+                1 + uninstantiated_attachments - self.atlas_config.max_uninstanciated_attachments;
+            self.evict_k_oldest_uninstantiated_attachments(to_delete)?;
```

Comment:
Won't this cause a lot of thrashing?  Suppose `self.atlas_config.max_uninstantiated_attachments` is 1024.  If I upload the 1025th uninstantiated attachment, won't this cause the node to delete all 1025 uninstantiated attachments, even if they were all otherwise valid and destined for inclusion?

Reply:
I think what you want to do here is just evict the oldest attachment, in order to make space for this next one.  In addition to (or possibly instead of) using time, you might want to consider the transaction fee -- lower-fee-rate transactions have their attachments evicted before higher-fee-rate ones.  Open to discussion.

Reply:
> Won't this cause a lot of thrashing? 
It's the other way, as shown in a test, if we receive a 1025th uninstantiated attachment, we delete the oldest one.